MassSpectrometry/MsDataFile.cs

```diff
@@ -86,18 +86,16 @@ public static int TopNpeakHelper(ref double[] intensities, ref double[] mArray,
         public static void WindowModeHelper(ref double[] intensities, ref double[] mArray, IFilteringParams filteringParams, double scanRangeMinMz, double scanRangeMaxMz, double? WindowMaxNormalizationToValue = null)
         {
             Array.Sort(mArray, intensities);
-
+            const double littleShift = 0.000000001;
             List<MzPeak> mzIntensites = new List<MzPeak>();
             List<MzPeak> mzIntensites_reduced = new List<MzPeak>();
+            
 
             //make MzPeaks of each mz/intensity pair and create a list of pairs for everything with intensity above the minimum cutoff
+            double maximumIntensityInArray = intensities.Max();
             for (int i = 0; i < mArray.Length; i++)
             {
-                if (filteringParams.MinimumAllowedIntensityRatioToBasePeakM.HasValue && (intensities[i] > (intensities.Max() * filteringParams.MinimumAllowedIntensityRatioToBasePeakM.Value)))
-                {
-                    mzIntensites.Add(new MzPeak(mArray[i], intensities[i]));
-                }
-                else if (!filteringParams.MinimumAllowedIntensityRatioToBasePeakM.HasValue)//don't filter on intensity and keep everything
+                if ((!filteringParams.MinimumAllowedIntensityRatioToBasePeakM.HasValue) || (filteringParams.MinimumAllowedIntensityRatioToBasePeakM.HasValue && (intensities[i] > (maximumIntensityInArray * filteringParams.MinimumAllowedIntensityRatioToBasePeakM.Value))))
```

Comment:
I like this change, but we don't need to check if it has a value in the second half of the "OR", because if it didn't have a value, we wouldn't have made it to the second half.

Reply:
done